com.unity.netcode.gameobjects/Runtime/Connection/NetworkConnectionManager.cs

```diff
@@ -347,7 +347,7 @@ internal ulong TransportIdCleanUp(ulong transportId)
             // When this happens, the client will not have an entry within the m_TransportIdToClientIdMap or m_ClientIdToTransportIdMap lookup tables so we exit early and just return 0 to be used for the disconnect event.
             if (!LocalClient.IsServer && !TransportIdToClientIdMap.ContainsKey(transportId))
             {
-                return 0;
+                return NetworkManager.LocalClientId;
```

Comment:
I am curious... if we are in the state as indicated by the comment above, how would LocalClientId ever be set? Wouldn't it still just be 0? I'm not sure returning a default-initialized variable that was never set is actually valid here.